bin/varnishd/cache/cache_req_fsm.c

```diff
@@ -382,6 +382,7 @@ cnt_synth(struct worker *wrk, struct req *req)
 
 	req->objcore = HSH_Private(wrk);
 	CHECK_OBJ_NOTNULL(req->objcore, OBJCORE_MAGIC);
+	req->objcore->flags &= ~OC_F_BUSY;
```

Comment:
This looks wrong to me. My understanding of `HSH_Private()` and the private objhead is that private objects should go through the same lifecycle as referenced objects, and in fact the synth object _is_ busy before it has had its body written.
I also do not understand the notion of "they will never acquire contents from a busyobj". There is no `ObjGetSpace()` without a busy object. Do we have different semantics of the flag in mind?

Reply:
Sorry for the confusion, by that I meant that a req task will never acquire contents from the busyobj of a fetch task. So there is no fetch task to "unbusy" the objcore and signal the req task and other clients. Maybe I meant a boc? I don´t remember, new year's eve happened meanwhile.

Reply:
My main point is really that, no matter what might use the busy flag or not, I think it is helpful to keep the same state transitions for private and cached objects. This was, I think, a relevant achievement of 68a424092d589932c782c9514d340f1752f094a5, and I think that we should not go back to special casing private objects in another way.

Reply:
Private (pass) objects have always gone through the same life cycle as regular fetch (miss, bgfetch) objects, in all iterations of this pull request (and the previous one).

The difference I'm drawing here is between objects that can keep a request busy during lookup, and objects that can never be looked up. We use the busy flag during lookup to decide when to park requests on a waiting list, and the related counters are called `MAIN.busy_*`. So `OC_F_BUSY` really is meant to coordinate requests with their waiting lists. The flag is also used to coordinate the non-waiting request task that triggers the fetch task, and in that case there is no difference between "private and cached objects" (quoting you) in terms of lifecycle.

However, `synth` or `req.body` objects check none of the proverbial request lifecycle boxes. I tried to better reflect that in the current patch series.

Reply:
> The flag is also used to coordinate the non-waiting request task that triggers the fetch task

Actually, I'm being imprecise here so let me correct myself. The boc is used to coordinate the fetch task with the request task that triggered it. And depending on whether we `do_stream` or not, we drop the flag sooner or later to wake up both requests parked on the waiting list, and the request blocking on the boc.

So the request and fetch tasks coordinates via a boc, and when we reach the point where requests can resume progress we coordinate the boc with the busy flag (to drop the latter).

The current patch series makes the boc responsible of this indirect coordination, instead of having all call sites duplicate the logic.